### PR TITLE
Add a dedicated is_none function.

### PR DESCRIPTION
**Pull request by LaurentMazare**
There are currently two ways to check whether a python value is none, one can either use caml structural equality with `Py.none` or get the python type with `Py.Type.get` and compare it to `None`.

This PR adds a very simple `is_none` function.
- Compared to structural equality, it feels more robust: when the none value is the result of some python function it's difficult to understand whether structural equality will work properly.
- Compared to `Py.Type.get`, calling `is_none` is a bit more efficient as it avoids various other tests.

### CODE REVIEW

File: py.ml [hunk 1]
```diff
@@ -902,6 +902,8 @@ module Type = struct
   let is_subtype a b =
     bool_of_int (Pywrappers.pytype_issubtype a b)
 
+  external is_none: pyobject -> bool = "pytype_is_none"
```
Comment (thierry-martinez) [inline py.ml]: I would have expected the name `pyobject_is_none` for the C stub. Is there a particular reason for `pytype`?

File: py.mli [hunk 2]
```diff
@@ -1817,6 +1817,9 @@ module Type: sig
   (** Wrapper for
       {{:https://docs.python.org/3/c-api/type.html#c.PyType_IsSubtype} PyType_IsSubtype} *)
 
+  val is_none: Object.t -> bool
+  (** [is_none o] returns true if the Python object [o] is None. *)
```
Comment (thierry-martinez) [inline py.mli]: `returns [true]`, `is [None]` (with square brackets)

**Comment by thierry-martinez** [comment]
Thank you very much for this PR. Indeed, I think such a function is useful indeed: it is clearer to write for instance `List.exists Py.is_none` than `List.exists (( = ) Py.none)`. However, structural equality should not be less robust: (1) structural equality calls `pycompare` which should properly compare any Python objects, (2) even physical equality should work in any case, because the only way to build a `pyobject` is to call `pyml_wrap`, which handles `None` (as well as other particular values) specially. This remark does not remove the interest of having a dedicated function of course.

**Comment by thierry-martinez** [comment]
I wonder if `let is_none v  = v == none` wouldn't do the job, while being even more efficient and easier to maintain. Do you have a test case where the physical equality wouldn't work? (In this case, it would perhaps require a fix.)

**Comment by LaurentMazare** [comment]
I was afraid that something like `Py.Run.eval "None"` would not work as the returned ocaml value may be different from the one created by `pynone` but actually they seem to be the same. I'm not really sure what makes this work and whether it's robust or not, my feeling was that you should compare the C pointers and not the ocaml ones.

**Comment by LaurentMazare** [comment]
Just got to read properly `pyml_wrap` and `pyml_unwrap` and indeed I agree that this should ensure that physical equality works well so I've switched to this as per your suggestion.
I also added a test case where `None` is generated via `Py.Run.eval` just in case.

**Comment by thierry-martinez** [comment]
I documented the guarantees that [Py.Object.t] offer for structural and physical equalities, and added [Py.is_null], [Py.Bool.{is_true,is_false}] and [Py.Tuple.is_empty] as well. I moved [Py.Type.is_none] to [Py.is_none]: it seems more logical to me that [Py.none] and [Py.is_none] are in the same level.

**Comment by thierry-martinez** [comment]
Thank you!

**Comment by LaurentMazare** [comment]
Cool, thanks for adding all these + taking the time to detail the implementation guarantees!
